packages/@sanity/validation/src/validateDocument.ts

```diff
@@ -59,17 +59,44 @@ export default async function validateDocument(
     return []
   }
 
-  try {
-    return await validateItem({
-      client,
-      schema,
-      parent: undefined,
-      value: doc,
-      path: [],
-      document: doc,
-      type: documentType,
-      getDocumentExists: context?.getDocumentExists,
+  const client = getClient({apiVersion: '2021-06-07'})
+  const validationOptions: ValidateItemOptions = {
+    client,
+    getClient,
+    schema,
+    parent: undefined,
+    value: doc,
+    path: [],
+    document: doc,
+    type: documentType,
+    getDocumentExists: context?.getDocumentExists,
+  }
+
+  // <TEMPORARY UGLY HACK TO PRINT DEPRECATION WARNINGS ON USE>
+  /* eslint-disable no-proto */
+  const wrappedClient = client as any
```

Comment:
Extract to a function to it easier to see the start and end of the hack?

Reply:
That would have been better. It is going away in the next release so will leave it as-is, but noted.